### PR TITLE
DIAG: Add CLI support for Diagnostic DIAG_GET.rsp message.

### PR DESCRIPTION
**Pull request by LuDuda**
This PR proposes a simple solution for #1055.

### CODE REVIEW

File: include/openthread.h [hunk 1]
```diff
@@ -1895,6 +1895,29 @@ OTAPI ThreadError OTCALL otGetParentInfo(otInstance *aInstance, otRouterInfo *aP
...
 
 /**
+ * This function pointer is called when an DIAG_GET.rsp is received.
+ *
+ * @param[in]  aMessage      A pointer to the message buffer containing the received DIAG_GET.rsp payload.
+ * @param[in]  aMessageInfo  A pointer to the message info for @p aMessage.
+ * @param[in]  aContext      A pointer to application-specific context.
+ *
+ */
+typedef void (*otReceiveDiagnosticGetCallback)(otMessage aMessage, const otMessageInfo *aMessageInfo,
```
Comment (jwhui) [inline include/openthread.h]: For request-response interactions, does it make more sense to include the callback/context as part of the request (i.e. `otSendDiagnosticGet()` in this scenario)?

Comment (LuDuda) [inline include/openthread.h]: The intention was to prepare the API for future implementation of DIAG_GET.qry/.ans which is intended for multicast messages, that i am going to add later on. However its not in the place now, so i can change API to add the last parameter of otSendDiagnosticGet to be of otCoapResponseHandler type (+context).

Comment (jwhui) [inline include/openthread.h]: Ah, yes, you had mentioned that in #1055.

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -2662,5 +2664,35 @@ void Interpreter::ProcessNetworkDiagnostic(int argc, char *argv[])
...
+{
+    static_cast<Interpreter *>(aContext)->HandleDiagnosticGetResponse(*static_cast<Message *>(aMessage),
+                                                                      *static_cast<const Ip6::MessageInfo *>(aMessageInfo));
+}
+
+void Interpreter::HandleDiagnosticGetResponse(Message &aMessage, const Ip6::MessageInfo &)
+{
+    sServer->OutputFormat("DIAG_GET.rsp: ");
+
+    uint8_t buf[16];
```
Comment (jwhui) [inline src/cli/cli.cpp]: nit: prefer to put variable declarations at top of containing scope (missed this in my initial review).

Comment (LuDuda) [inline src/cli/cli.cpp]: Thanks! Fixed in the next commit.

**Pull request merged**